src/components/LibrariesList.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { PanelGroup } from "react-bootstrap";
+import { Store, Reducer } from "redux";
+import { State } from "../reducers/index";
+import { LibrariesData, LibraryData } from "../interfaces";
+import { connect } from "react-redux";
+import ActionCreator from "../actions";
+import LibrariesListItem from "./LibrariesListItem";
+
+export interface LibrariesListStateProps {
+  libraries?: LibrariesData;
+}
+
+export interface LibrariesListOwnProps {
+  store?: Store<State>;
+  csrfToken?: string;
+}
+
+export interface LibrariesListDispatchProps {
+  fetchData: () => Promise<LibrariesData>;
+}
+
+export interface LibrariesListState {
+  activeKey: string;
+}
+
+export interface LibrariesListProps extends LibrariesListStateProps, LibrariesListOwnProps, LibrariesListDispatchProps {};
+
+export class LibrariesList extends React.Component<LibrariesListProps, LibrariesListState> {
+
+  constructor(props: LibrariesListProps) {
+    super(props);
+    this.state = { activeKey: "" };
+    this.isActive = this.isActive.bind(this);
+    this.select = this.select.bind(this);
+  }
+
+  isActive(library: LibraryData) {
+    return this.props.libraries.libraries.indexOf(library) === parseInt(this.state.activeKey);
+  }
+
+  select(idx: string) {
+    this.setState({ activeKey: idx});
+    this.props.fetchData();
+  }
+
+  render(): JSX.Element {
+    return(
+      <PanelGroup accordion activeKey={this.state.activeKey}>
+        { this.props.libraries &&
+          this.props.libraries.libraries.map(library =>
```

Comment:
I think you can change this to `.map((library, idx) =>` instead of having to compute the index for every library, which could be slow if there are a lot.

Reply:
I agree, `.map` would be better. Also, use `index` instead of `idx` to be clearer.